tiledb/sm/storage_manager/open_array.h

```diff
@@ -174,9 +180,14 @@ class OpenArray {
   /*         PRIVATE ATTRIBUTES        */
   /* ********************************* */
 
-  /** The array schema. */
+  /** The latest array schema. */
   ArraySchema* array_schema_;
 
+  /**
+   * A map of all array_schemas
+   */
+  std::map<std::string, ArraySchema*> array_schemas_;
```

Comment:
Instead of `ArraySchema*` can we use a `shared_ptr` here (and else where)?

Reply:
Should we replace raw pointer of ArraySchema* everywhere? I prefer to replacing all raw pointers with tdb_shared_ptr or tdb_unique_ptr. If our future plan is to remove all raw pointers, I can start to replace all ArraySchema*

Reply:
switched ArraySchema* to tdb_shared_ptr<ArraySchema> only for the new unorderd_map(like array_metadata_). In FragmentMetadata, raw pointer is still used. I tried to change the raw pointer to shared_ptr in FragmentMetadata, but found that we also need to change it in writer and other classes.